src/ui/components/home/home_activity.rs

```diff
@@ -14,56 +17,80 @@ pub async fn list_posts(form: &GetPosts) -> Result<GetPostsResponse, LemmyAppErr
 
 #[component]
 pub fn HomeActivity() -> impl IntoView {
-  let query = use_query_map();
-  let page = move || {
-    query
-      .with(|q| q.get("page").and_then(|page| page.parse::<i64>().ok()))
-      .unwrap_or(1)
-  };
+  let _query = use_query_map();
+
+  let error = create_rw_signal::<Option<String>>(None);
+
+  let authenticated = expect_context::<RwSignal<bool>>();
+
+  let next_page_cursor = create_rw_signal::<Option<PaginationCursor>>(None);
+  let page_cursor = create_rw_signal::<Option<PaginationCursor>>(None);
 
-  let authenticated = use_context::<RwSignal<bool>>().unwrap_or(create_rw_signal(false));
+  let prev_cursor_stack = create_rw_signal::<Vec<Option<PaginationCursor>>>(vec![]);
```

Comment:
Having just done this for lemmyUI, unfortunately you pretty much need to rely on the browser back button, because keeping a stack won't work. IE hitting the `previous` button actually might go forward a page, since that might have been the last page you visited.

Reply:
:laughing: yeah i have tried to replicate a back button there, browser back would be much easier although doesn't work without JS?

Reply:
I'd have to test but I don't see why the leptos router wouldn't work without js.